### PR TITLE
Usaco 764

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-764.mdx [hunk 1]
```diff
@@ -146,4 +144,44 @@ public class shuffle {
...
+
+```python
+from collections import deque
+
+inp = open("shuffle.in", "r")
+
+n = int(inp.readline())
+
+cnt = [0] * n
+a = list(map(int, inp.readline().split()))
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:157]: i'd use a `with` block
also please rename `cnt` to something else

File: solutions/usaco-764.mdx [hunk 2]
```diff
@@ -146,4 +144,44 @@ public class shuffle {
...
+
+for i in range(n):
+	a[i] -= 1
+	cnt[a[i]] += 1
+
+ans = n
+q = deque()
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:164]: better names

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-764.mdx [hunk 3]
```diff
@@ -53,26 +53,36 @@ void setIO(string fileName = ""){
...
 int main(){
 	setIO("shuffle");
-	ll n; cin >> n;
-	vl cnt(n), a(n);
+	ll n;
+	cin >> n;
+
+	vl a(n);
+	vl cows_after_shuffle(n);
 	FOR(i, 0, n){
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:61]: could you remove macros

File: solutions/usaco-764.mdx [hunk 4]
```diff
@@ -146,4 +154,44 @@ public class shuffle {
 
 </JavaSection>
 
+<PySection>
+
+```python
+from collections import deque
+
+with open("shuffle.in", "r") as input_file:
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:162]: i'd use single quotes for python, don't really care about this though

Comment (876pol) [reply solutions/usaco-764.mdx]: oop too used to c++

File: solutions/usaco-764.mdx [hunk 5]
```diff
@@ -2,97 +2,75 @@
...
-	cout << ans << endl;
+int main() {
+    freopen("shuffle.in", "r", stdin);
+    freopen("shuffle.out", "w", stdout);
+
+    ll n;
+    cin >> n;
+    vector<ll> a(n);
+    vector<ll> cows_after_shuffle(n);
+    for (int i = 0; i < n; i++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-764.mdx:31]: tabs

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**